### PR TITLE
Fix docstrings

### PR DESCRIPTION
**Pull request by atomassi**
This PR fixes some of the parameters in the docstrings (typos or missing) + remove a couple of unused variables.

### CODE REVIEW

File: networkx/generators/trees.py [hunk 1]
```diff
@@ -163,6 +163,8 @@ def random_tree(n, seed=None, create_using=None):
     seed : integer, random_state, or None (default)
         Indicator of random number generation state.
         See :ref:`Randomness<randomness>`.
+    create_using : NetworkX graph instance, optional
+        Return graph of this type.
```
Comment (dschult) [inline networkx/generators/trees.py:167]:
```suggestion
    create_using : NetworkX graph constructor, optional (default=nx.Graph)
        Graph type to create. If graph instance, then cleared before populated.
```

File: networkx/readwrite/gexf.py [hunk 2]
```diff
@@ -50,6 +50,8 @@ def write_gexf(G, path, encoding="utf-8", prettyprint=True, version="1.2draft"):
        Encoding for text data.
     prettyprint : bool (optional, default: True)
        If True use line breaks and indenting in output XML.
+    version: string (optional default: '1.2draft')
+       The version of GEXF to be used for nodes attrbutes checking
```
Comment (dschult) [inline networkx/readwrite/gexf.py:54]:
```suggestion
    version: string (optional, default: '1.2draft')
       The version of GEXF to be used for nodes attributes checking
```

File: networkx/algorithms/shortest_paths/astar.py [hunk 3]
```diff
@@ -157,6 +157,12 @@ def astar_path_length(G, source, target, heuristic=None, weight="weight"):
        from the a node to the target.  The function takes
        two nodes arguments and must return a number.
 
+    weight : string or function
+        If it is callable, `weight` itself is returned. If it is a string,
+        it is assumed to be the name of the edge attribute that represents
+        the weight of an edge. In that case, a function is returned that
+        gets the edge weight according to the specified edge attribute.
+
```
Comment (dschult) [inline networkx/algorithms/shortest_paths/astar.py:165]:
```suggestion
    weight : string or function
       If this is a string, then edge weights will be accessed via the
       edge attribute with this key (that is, the weight of the edge
       joining `u` to `v` will be ``G.edges[u, v][weight]``). If no
       such edge attribute exists, the weight of the edge is assumed to
       be one.
       If this is a function, the weight of an edge is the value
       returned by the function. The function must accept exactly three
       positional arguments: the two endpoints of an edge and the
       dictionary of edge attributes for that edge. The function must
       return a number.
```

**Review by dschult — APPROVED** [review]
Thanks for this...   I usually find these types of PRs too cumbersome to review well, but this was pretty straightforward.  Thanks for that and thanks for finding all these goofs.

How did you find them? Is there a tool, or did you just look through the docs/code?

I approve these changes. -- but consider the suggestions below to match existing docstrings elsewhere in the file/package.

**Comment by atomassi** [comment]
Thank you for spending time reviewing it and for the suggested changes, they totally make sense.  
Agree, with you, since this was mostly about docstrings I've not considered it worth to create separates pull requests.
I've used PyCharm, which has very nice feature [inspect code](https://www.jetbrains.com/help/pycharm/inspections-settings.html) to find typos, type checking violations, incorrect docstrings, and so on.

File: networkx/readwrite/multiline_adjlist.py [hunk 4]
```diff
@@ -141,6 +141,10 @@ def write_multiline_adjlist(G, path, delimiter=" ", comments="#", encoding="utf-
     ----------
     G : NetworkX graph
 
+    path : string or file
+       Filename or file handle to read.
```
Comment (rossbar) [inline networkx/readwrite/multiline_adjlist.py:145]:
```suggestion
       Filename or file handle to write to.
```

File: networkx/readwrite/nx_yaml.py [hunk 5]
```diff
@@ -29,9 +29,9 @@ def write_yaml(G_to_be_yaml, path_for_yaml_output, **kwds):
 
     Parameters
     ----------
-    G : graph
+    G_to_be_yaml : graph
```
Comment (rossbar) [inline networkx/readwrite/nx_yaml.py:32]: +1 for fixing, though these functions are deprecated and will be removed soon. Nice catch nonetheless!

**Review by rossbar — COMMENTED** [review]
LGTM save for one read/write wording issue. Thanks @atomassi !

**Pull request merged**